### PR TITLE
Fix/id type support

### LINKED ISSUE #53: PostAsync entity id check fails for Guid type
Since `Guid` type does not have null value, it is always initialized to `Guid.Empty` and [this check](https://github.com/Research-Institute/json-api-dotnet-core/blob/master/src/JsonApiDotNetCore/Controllers/JsonApiController.cs#L142) in `JsonApiController` fails (string value of Guid.Empty is "000-000...").

Looking around I was not able to find any generic way to check for empty values. So I would suggest adding method `IsIdEmpty` or something similar to `IIdentifiable` with default implementation maybe as you have it right now, or simply checking for null value. Then users could implement that method with whatever logic their ids have.

If that seems reasonable for you, I might make a pull request with that change tomorrow.

### PR DESCRIPTION
**Pull request by jaredcnance**
Closes #53

### CODE REVIEW

**Comment by jaredcnance** [comment]
@jfhs this **should** fix the issues. I will be writing tests to validate it in the morning. Let me know what you think

**Comment by jfhs** [comment]
@jaredcnance Looks good. However I checked [specification](http://jsonapi.org/format/#crud-creating-client-ids), and it says that server may accept client generated ids which is explicitly forbidden in current implementation. That is not the same issue you are solving by this PR, but it requires change in same place, so maybe you can do it all at once.

I would suggest adding a flag in controller for disabling this verification, something like `bool AllowClientGeneratedId` which will be false by default, and can be re-enabled by users. Although that's not required for my current use-case, I think in general that would be good to have.

**Comment by jaredcnance** [comment]
You're right in that a server MAY support client generated ids. I'll include a test this morning that checks whether or not it will work without much modification. I suspect this may not be so simple, but I'll dig into it.

File: src/JsonApiDotNetCore/Serialization/JsonApiDeSerializer.cs [hunk 1]
```diff
@@ -98,29 +111,68 @@ private static object DataToObject(DocumentData data, IJsonApiContext context)
...
+
+                var newValue = data["id"];
+                var convertedValue = TypeHelper.ConvertType(newValue, entityProperty.PropertyType);
+                entityProperty.SetValue(entity, convertedValue);
             }
 
             return entity;
         }
 
+        private static object _setHasManyRelationship(object entity,
```
Comment (jaredcnance) [inline src/JsonApiDotNetCore/Serialization/JsonApiDeSerializer.cs]: Closes #52

**Comment by jaredcnance** [comment]
You can get these updates on [MyGet](https://www.myget.org/feed/research-institute/package/nuget/JsonApiDotNetCore) (version 1.1.1-alpha1-0069)

**Pull request merged**